tests/test_on_data/test_format_converters/test_miniscope_converter.py

```diff
@@ -0,0 +1,134 @@
+import shutil
+import tempfile
+from datetime import datetime
+from pathlib import Path
+from unittest import TestCase
+from warnings import warn
+
+from pynwb import NWBHDF5IO
+from pynwb.image import ImageSeries
+from pynwb.ophys import OnePhotonSeries
+
+from neuroconv import ConverterPipe, NWBConverter
+from neuroconv.converters import MiniscopeConverter
+from tests.test_on_data.setup_paths import OPHYS_DATA_PATH
+
+
+class TestMiniscopeConverter(TestCase):
+    @classmethod
+    def setUpClass(cls) -> None:
+        cls.folder_path = str(OPHYS_DATA_PATH / "imaging_datasets" / "Miniscope" / "C6-J588_Disc5")
+        cls.converter = MiniscopeConverter(folder_path=cls.folder_path)
+        cls.test_dir = Path(tempfile.mkdtemp())
+
+        cls.device_name = "Miniscope"
+        cls.device_metadata = dict(
+            name=cls.device_name,
+            compression="FFV1",
+            deviceType="Miniscope_V3",
+            frameRate="15FPS",
+            framesPerFile=1000,
+            gain="High",
+            led0=47,
+        )
+
+        cls.behavcam_name = "BehavCam2"
+        cls.behavcam_metadata = dict(
+            name=cls.behavcam_name,
+            compression="MJPG",
+            deviceType="WebCam-1920x1080",
+            framesPerFile=1000,
+            ROI={"height": 720, "leftEdge": 0, "topEdge": 0, "width": 1280},
+        )
+
+        cls.image_series_name = "BehavCamImageSeries"
+        cls.photon_series_name = "OnePhotonSeries"
+
+    @classmethod
+    def tearDownClass(cls) -> None:
+        try:
+            shutil.rmtree(cls.test_dir)
+        except PermissionError:
+            warn(f"Unable to cleanup testing data at {cls.test_dir}! Please remove it manually.")
+
+    def test_converter_metadata(self):
+        metadata = self.converter.get_metadata()
+        self.assertEqual(
+            metadata["NWBFile"]["session_start_time"],
+            datetime(2021, 10, 7, 15, 3, 28, 635),
+        )
+        self.assertDictEqual(
+            metadata["Ophys"]["Device"][0],
+            self.device_metadata,
+        )
+
+        self.assertDictEqual(
+            metadata["Behavior"]["Device"][0],
+            self.behavcam_metadata,
+        )
+
+    def test_run_conversion(self):
+        nwbfile_path = str(self.test_dir / "test_miniscope_converter.nwb")
+        self.converter.run_conversion(nwbfile_path=nwbfile_path)
+
+        self.assertNWBFileStructure(nwbfile_path=nwbfile_path)
+
+    def test_run_conversion_updated_metadata(self):
```

Comment:
This test does a great job showcasing the metadata control of this converter; can you add some extra tests showcasing the `conversion_options` and its schema as well? (if there are any... showcase how it's empty otherwise?)

In particular, how it appears when nested inside another NWBConverter/ConverterPipe